implementation/src/main/java/io/smallrye/mutiny/helpers/test/UniAssertSubscriber.java

```diff
@@ -341,6 +342,17 @@ public UniAssertSubscriber<T> assertItem(T expected) {
         return this;
     }
 
+    /**
+     * Assert that the {@link io.smallrye.mutiny.Uni} has received an item.
+     * @param function the function to apply the item to assert on
+     * @param expected the expected item
+     * @param <R> the return type of the item
+     */
+    public <R> void assertItem(Function<T, R> function, R expected) {
```

Comment:
You could actually just delegate to `assertItem(expected)`

Reply:
good point, thanks!

Reply:
actually, I need to `apply` the function.